project/src/js/ui/tournament/tournamentView.js

```diff
@@ -0,0 +1,156 @@
+import * as utils from '../../utils';
+import h from '../helper';
+import { header as headerWidget, backButton, empty, pad} from '../shared/common';
+import layout from '../layout';
+import m from 'mithril';
+
+export default function view(ctrl) {
+  const headerCtrl = utils.partialf(headerWidget, null,
+    backButton(ctrl.tournament().fullName)
+  );
+  const bodyCtrl = tournamentBody.bind(undefined, ctrl);
+
+  return layout.free(headerCtrl, bodyCtrl, empty, empty);
+}
+
+function tournamentBody(ctrl) {
+  let data = ctrl.tournament();
+  if (!data) return null;
+
+  let body = null;
+  if (data.isFinished)
+    body = tournamentContentFinished(data);
+  else if (!data.isStarted)
+    body = tournamentContentCreated(data);
+  else
+    body = tournamentContentStarted(data);
+
+  return (m('.tournamentContainer', body));
+}
+
+function tournamentContentFinished(data) {
+  return (
+    <div>
+      { tournamentHeader(data, null, null)}
+      { tournamentLeaderboard(data, true) }
+    </div>
+  );
+}
+
+function tournamentContentCreated(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToStart, 'Starts in:')}
+      { tournamentLeaderboard(data, false) }
+    </div>
+  );
+}
+
+function tournamentContentStarted(data) {
+  return (
+    <div>
+      { tournamentHeader(data, data.secondsToFinish, '')}
+      { tournamentLeaderboard(data, false) }
+      { tournamentFeaturedGame(data) }
+    </div>
+  );
+}
+
+function tournamentHeader(data, time, timeText) {
+  let variant = variantInfo(data);
+  let control = timeControl(data);
+  return (
+    <div className='basicTournamentInfo'>
+      <strong> {variant + ' • ' + control + ' • ' + data.minutes + 'M' } </strong>
+      <div className='timeInfo'>
+        <strong> {timeInfo(time, timeText)} </strong>
+      </div>
+    </div>
+  );
+}
+
+function variantInfo(data) {
+  let variant = data.variant;
+  if(variant === 'standard') {
+    if(data.schedule)
+      variant = data.schedule.speed;
+    else if(data.position)
+      variant = data.position.name;
+    else
+      variant = '';
+  }
+
+  while(variant.length > 12) {
```

Comment:
Here I'm not sure to understand what is done just at looking at the code. Generally prefer using functionnal style (`map`, `filter`, etc) and no `for` or `while` loops. 
Its' important for readability, maintainability and to ensure consistency within the project.